src/os/filestore/FileStore.cc

```diff
@@ -718,6 +718,14 @@ int FileStore::statfs(struct store_statfs_t *buf0)
   }
   buf0->total = buf.f_blocks * buf.f_bsize;
   buf0->available = buf.f_bavail * buf.f_bsize;
+  // Adjust for writes pending in the journal
+  if (journal) {
+    uint64_t estimate = journal->get_journal_size_estimate();
```

Comment:
@liupan1111 there are two cases:

1. journal and data are co-located in the same filesystem (fs for short), in which case, `FileStore::statfs()` should return the space still available for the new data. so 
```c
fs.available = fs.all - data.used - journal.max_size // the space used by journal is pre-allocated
data.all = fs.all - journal.max_size
buf0.available = fs.available - journal.used
```
2. journal and data are living in different filesystems, in which case, 
```c
fs.available = fs.all - data.used
data.all = fs.all
buf0.available = fs.available - journal.used
```

so i think the change is correct.